src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -264,7 +297,36 @@ class UserService(
     return AuthorisableActionResult.Success(user)
   }
 
-  fun updateUser(
+  fun updateUserEntity(
+    user: UserEntity,
+    staffDetail: StaffDetail,
+    forService: ServiceName,
+  ): UserEntity {
+    user.name = staffDetail.name.toDeliusName()
+    user.email = staffDetail.email
+    user.telephoneNumber = staffDetail.telephoneNumber
+    user.deliusStaffCode = staffDetail.code
+    user.teamCodes = staffDetail.teamCodes()
+
+    user.probationRegion =
+      probationAreaProbationRegionMappingRepository
+        .findByProbationAreaDeliusCode(staffDetail.probationArea.code)
+        ?.probationRegion!!
```

Comment:
? followed by !! seems like it could lead to null pointers

In the prior version of this code if we couldn't determine the probation region on update we would leave it set to its current value. I don't love that behaviour tbh but i think it was required because of the 'assignDefaultRegionToUsersWithUnknownRegion' flag that is used when creating users. Ideally we should remove that flag (and/or use the exact same logic when creating/updating users which would cut down in duplication), but in the meantime it may be to not set this if a probation region can't be resolved?